mne/surface.py

```diff
@@ -1689,38 +1697,80 @@ def marching_cubes(image, level, smooth=0):
 
     if image.ndim != 3:
         raise ValueError(f'3D data must be supplied, got {image.shape}')
+    # force double as passing integer types directly can be problematic!
+    image_shape = image.shape
     data_vtk = numpy_support.numpy_to_vtk(
-        image.ravel(), deep=True, array_type=VTK_DOUBLE
-    )
+        image.ravel(), deep=True, array_type=VTK_DOUBLE)
+    del image
+    if not _is_numeric(level):
+        level = np.array(level)
+        if level.ndim != 1 or level.size == 0 or level.dtype.kind not in 'ui':
+            raise TypeError(
+                'level must be non-empty numeric or 1D array-like of int, '
+                f'got {level.ndim}D array-like of {level.dtype} with '
+                f'{level.size} elements')
+        mc = vtkDiscreteFlyingEdges3D()
+        discrete = True
+    else:
+        mc = vtkMarchingCubes()
+        discrete = False
+        level = [float(level)]
     # create image
     imdata = vtkImageData()
-    imdata.SetDimensions(image.shape)
+    imdata.SetDimensions(image_shape)
     imdata.SetSpacing([1, 1, 1])
     imdata.SetOrigin([0, 0, 0])
     imdata.GetPointData().SetScalars(data_vtk)
 
     # compute marching cubes
-    mc = vtkMarchingCubes()
+    mc.SetNumberOfContours(len(level))
+    for li, lev in enumerate(level):
+        mc.SetValue(li, lev)
     mc.SetInputData(imdata)
-    mc.SetValue(0, level)
-    mc.Update()
     out = mc
     if smooth:
-        filt = vtkWindowedSincPolyDataFilter()
-        filt.SetInputConnection(mc.GetOutputPort())
-        filt.SetNumberOfIterations(100)
-        filt.SetPassBand(1 - smooth)
-        filt.Update()
-        out = filt
-    polydata = out.GetOutput()
+        smoother = vtkWindowedSincPolyDataFilter()
+        smoother.SetInputConnection(mc.GetOutputPort())
+        smoother.SetNumberOfIterations(100)
+        smoother.BoundarySmoothingOff()
+        smoother.FeatureEdgeSmoothingOff()
+        smoother.SetFeatureAngle(120.0)
+        smoother.SetPassBand(1 - smooth)
+        smoother.NonManifoldSmoothingOn()
+        smoother.NormalizeCoordinatesOff()
+        out = smoother
+    out.Update()
 
     # get verts and triangles
-    verts = numpy_support.vtk_to_numpy(polydata.GetPoints().GetData())
-    triangles = numpy_support.vtk_to_numpy(
-        polydata.GetPolys().GetConnectivityArray()).reshape(-1, 3)
-    verts = np.flip(verts, axis=1)
-    triangles = np.flip(triangles, axis=1)
-    return verts, triangles
+    polydatas = list()
```

Comment:
data is already plural, something else datas is in `mne.viz.plot_alignment` too

Reply:
Yes but the object is `PolyData`, we sometimes just keep adding `s`'s to indicate more of something, e.g. `poss` as "positions-es"